changelog/std-array-asStatic.md

```diff
@@ -0,0 +1,25 @@
+Added `staticArray` to construct a static array from input array / range / CT range.
+
+The type of elements can be specified implicitly ($(D [1, 2].staticArray) of type `int[2]`)
+or explicitly ($(D [1, 2].staticArray!float) of type `float[2]`).
+When `a` is a range with length not known at compile time, the number of elements must be given as template argument
+(eg `myrange.staticArray!2`).
+Size and type can be combined, if source range elements are implicitly
+convertible to the requested element type (eg: `2.iota.staticArray!(long[2])`).
+When the range `a` is known at compile time, it can also be specified as a
+template argument to avoid having to specify the number of elements
+(eg: `staticArray!(2.iota)` or `staticArray!(double, 2.iota)`).
+
+Note: $(D fun([1, 2, 3].staticArray)) may be inefficient because of the copies involved.
```

Comment:
Ask Walter, not me. Because the comment was here already I thought there may be a reason for it and decided to leave it there..

Reply:
I would remove the note from the changelog as I think it's clear that in the general case static arrays need to be copied and such details belong into the detailed documentation of the function.

See also: https://godbolt.org/g/GEsh3E

Reply:
Oh, i didn't notice that static array was being fed to a function. I thought he meant that [1, 2, 3] might be unnecessarily copied when initializing. Sounded unlikely, but I thought that Timothee might know something I do not.

But since it means that feeding a static array to a function involves a copy, I will remove that.

But looking at that godbolt example, why does LDC make `bar` to copy the static array when passing it to `foo` when it does not use it for anything else? Can't it rely on `foo` to copy it if it mutates it?